app/src/main/java/com/example/taskyy/ui/screens/ReminderScreen.kt

```diff
@@ -816,3 +828,65 @@ fun SaveTitle(details: String, onEvent: (EditTextEvent) -> Unit) {
     )
 }
 
+@Composable
+fun CheckForNotificationPermissions() {
+    val context = LocalContext.current
+    var hasNotificationPermission by remember {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            mutableStateOf(
+                ContextCompat.checkSelfPermission(
+                    context,
+                    Manifest.permission.POST_NOTIFICATIONS,
+                ) == PackageManager.PERMISSION_GRANTED,
+            )
+        } else {
+            mutableStateOf(true)
+        }
+    }
```

Comment:
If a value never changes in Compose, there's no point of making it Compose state. You can just make it a normal `val`